source/mqtt_agent.c

```diff
@@ -746,6 +791,8 @@ static MQTTStatus_t resendPublishes( MQTTAgentContext_t * pMqttAgentContext )
 
             if( statusResult != MQTTSuccess )
             {
+                concludeCommand( pMqttAgentContext, pFoundAck->pOriginalCommand, statusResult, NULL );
+                ( void ) memset( pFoundAck, 0x00, sizeof( AckInfo_t ) );
                 LogError( ( "Error in resending publishes. Error code=%s\n", MQTT_Status_strerror( statusResult ) ) );
```

Comment:
Unrelated to the PR changes but I noticed the error log and thus am suggesting an improvement. The presence of "Error" in statement is redundant/
```suggestion
                LogError( ( "Unable tp resend publishes. Error code=%s\n", MQTT_Status_strerror( statusResult ) ) );
```